_posts/2022-09-07-redis-api-intro.adoc

```diff
@@ -0,0 +1,226 @@
+
+The `set` method emits the Redis `set` command and store a `Ranking` to the passed key.
+The command also configures the expiration time.
+So, after 10 seconds, the value is removed by Redis.
+As mentioned above, the Ranking instance is serialized into a JSON document.
+
+The `evict` method allows removing the stored value.
+Multiple commands can do this, such as the `del` or `getdel` (which also returned the stored value).
+
+For our application, we need something a bit more fancy.
+We want to check if we have a value in Redis.
+If so, uses that value, and if not, compute the value and store it.
+This is implemented in the `getOrSetIfAbsent`.
+
+Now, we can just use this cache to avoid the heavy computation on every call (check the https://github.com/cescoffier/quarkus-redis-demo/blob/main/src/main/java/me/escoffier/quarkus/supes/HeroService.java[HeroService] class to see the complete code):
+
+[source,java]
+----
+@Inject
+MyRedisCache cache;
+
+public Ranking getTopHeroes() {
+    return cache.getOrSetIfAbsent("top", () -> {
+                // Dumb approach, don't do this
+                return new Ranking(Hero.<Hero>listAll()
+                        .stream()
+                        .sorted((o1, o2) -> Integer.compare(o2.level, o1.level))
+                        .peek(h -> {
+                            // do something very long...
+                            nap();
+                        })
+                        .limit(3)
+                        .collect(Collectors.toList()));
+            });
+}
+----
+
+To run the application, just start `mvn quarkus:dev` and open your browser to http://localhost:8080:
+
+image::/assets/images/posts/redis-intro/screenshot.png[align="center"]
+
+To see the cache in action, check the time displayed on the page and refresh the page.
+Don't forget that the cached value is only valid for 10 seconds (as set in `MyRedisCache`).
+So, if you wait 10 seconds, it will recompute the result.
+
+Quarkus comes with a Redis Dev Service that automatically starts a Redis instance on your machine and configures the application. Note that you need to be able to run containers locally to use this feature.
+
+== Conclusion
+
+This post briefly introduces the new Redis API and demonstrates its usage with a cache implementation example.
+The complete code is available from this  https://github.com/cescoffier/quarkus-redis-demo[GitHub repo].
+
+The API supports many more features, such as geospatial data, pub/sub, and transactions, which could be used to improve the `getOrSetIfAbsent`  method.
+We will cover more advanced use cases in future posts.
+
+You can find more details about the new API in the:
+
+* https://quarkus.io/guides/redis[Quarkus Getting Started guide]
+* https://quarkus.io/guides/redis-reference[Quarkus Redis Reference guide]
+
+Also, the Quarkus team is working on integrating Redis as a cache implementation. So, eventually, you will just need to use `@CacheResult`, and let the magic happens.
```

Comment:
Can we add a link to a github issue/PR, might help with interested people tracking (see if the issue is resolved, released etc)?